### PR TITLE
Sync with community version

### PR DESCRIPTION
**Pull request by ericjoneshpe**
## Summary and Scope

Sync changes from HPE community version of xpmem minus any HPE pipeline build components. There were about 10 commits made to the internal HPE "community" repo that I've synced to this repo/branch. I also updated the RPM spec files to at least produce something usable.

## Issues and Related PRs

USS-2476

## Testing

Built on SLES15SP6 and RHEL9.4

Built RPMS on SLES15SP6

### Tested on:

VM

## Risks and Mitigations

Does not build under RHEL9.5. JIRA USS-2583 opened. 

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] Testing is appropriate and complete, if applicable

### CODE REVIEW

**Review by jamesclough-hpe — APPROVED** [review]

File: configure.ac [hunk 1]
```diff
@@ -36,27 +36,37 @@ AC_PROG_GCC_TRADITIONAL
...
+AC_SUBST([pkgconfigdir], [$with_pkgconfig_prefix])
 
 AC_PATH_KERNEL_SOURCE
 AC_KERNEL_CHECKS
 
 AC_CONFIG_FILES([Makefile
-                 cray-xpmem.pc
-                 module
+                 xpmem.pc
+                 module
```
Comment (jamesclough-hpe) [inline configure.ac:66]: NIT: Need to remove the space after 'module'

File: xpmem-dkms.spec [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Source: xpmem-2.6.5.tar.bz2
+Requires: dkms
+Provides: kmod(xpmem.ko)
+BuildArch: noarch
+
+%description
+XPMEM is a Linux kernel module that enables a process to map the
+memory of another process into its virtual address space. Source code
+can be obtained by cloning the Git repository, original Mercurial
+repository or by downloading a tarball from the link above.
```
Comment (deantroe) [inline xpmem-dkms.spec:23]: Let's edit this description like to match what was done in the xpmem-lib.spec file.

File: xpmem-kmod.spec [hunk 3]
```diff
@@ -24,21 +24,21 @@ can be obtained by cloning the Git repository, original Mercurial
 repository or by downloading a tarball from the link above.
```
Comment (deantroe) [inline xpmem-kmod.spec:24]: Let's edit this description like to match what was done in the xpmem-lib.spec file.

**Review by deantroe — APPROVED** [review]
Just made a few comments to fix up some description entries.

File: xpmem-dkms.spec (hunk 2 above)
Comment (ericjoneshpe) [reply xpmem-dkms.spec]: Fixed

**Pull request merged**